### ISSUE TITLE
Add support for ":1234" as hostport?

### ISSUE THREAD

**Issue by jcchavezs** [openzipkin/zipkin-go#89]
Currently there if using `NewEndpoint("name", ":1993")` does not provide the same effect as `&Endpoint{ServiceName: "name", Port: 1993}`, should they have the same behavour?

ping @basvanbeek

**Comment by basvanbeek** [comment]
Yes that would make sense. I guess it currently "guesses" localhost?

**Comment by adriancole** [comment]
not sure port without IP helps much. Why would we add this?

>

**Comment by jcchavezs** [comment]
In my case I don't know the IP at application level, only the port because
one do http.ListenAndServe(":1993", nil) so you only know the port by
default, in order to know the IP one might need to inject that value from
the outside (not sure it is worth it).

Den ons. 7. nov. 2018, 02:51 skrev Adrian Cole <notifications@github.com:

**Comment by adriancole** [comment]
oh ic. interesting.. yeah I guess you could decide to do like this or make
something like localPort to clarify this is a decoration not the complete
value.

In the java space, I've not really come across a case yet where you know
the port but not the IP. I think this is because we are getting it from the
socket itself.
https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/Span.java#L168

>
That doesn't mean it isn't valid, just from an API pov, tried not to have
people end up with a port and no ip address!

**Comment by adriancole** [comment]
The tracer should already have the link local IP address, you are just
clarifying it with a port. a port without an address is likely not
extremely useful, but maybe I'm wrong. I'm not trying to argue, just
saying that a dangling port isn't a great experience.
